### PR TITLE
fix auto closing pairs issue

### LINKED ISSUE #571: [Helm] `{{` Autocomplete adds extra `-}}` when `-` is typed
If I type `{{` code automatically adds `}}` and everything is fine. but if I write `-`, then code automatically adds `-}}` and I'll end up with `{{--}}}}` which is not fine.

### PR DESCRIPTION
**Pull request by ahmadalli**
fixes #571

### CODE REVIEW

File: language-configuration.json [hunk 1]
```diff
@@ -15,8 +15,9 @@
     ],
     // symbols that are auto closed when typing
     "autoClosingPairs": [
-        ["{{", "}}"],
```
Comment (zackijack) [inline language-configuration.json]: @ahmadalli I think you just need to remove the double bracket pairs because the single bracket will handle that.

Comment (ahmadalli) [reply language-configuration.json]: yeah that's doable too

File: language-configuration.json [hunk 2]
```diff
@@ -15,8 +15,8 @@
...
         ["{", "}"],
+        ["{{-", "-"],
```
Comment (zackijack) [inline language-configuration.json]: Again, this is just my assumption, why not just add `["-", "-"]` instead of `["{{-", "-"]`?

Comment (ahmadalli) [reply language-configuration.json]: we don't want to have `--` for every `-` user types. do we?

Comment (zackijack) [reply language-configuration.json]: Ah I see

**Comment by itowlson** [comment]
@ahmadalli Testing this, when I type `{{-` I end up with `{{--}}`.  This seems to be one more `-` than is desired - don't you want it to be `{{-}}`?  If you're happy that it's doing the right thing then I'll merge but this doesn't look quite right to me...?

**Comment by ahmadalli** [comment]
@itowlson both `{{- code }}` and `{{- code -}}` are valid with different use cases. the decision of the contributors was having `-}}` when user typed `{{-` but the functionality was faulty (what I described on https://github.com/Azure/vscode-kubernetes-tools/issues/571#issue-457501899) so it was removed to fix #571. this PR fixes both the original issue of #571 and the new issue reported on #571 and #620

**Comment by buddyledungarees** [comment]
For anyone else that runs into this autocomplete issue, this MR didn't change the behavior to how I prefer it so my workaround is to overwrite the packaged language-configuration.json under `~/.vscode/ms-kubernetes-tools.vscode-kubernetes-tools-{version}/` with the following setting:

```
{
    "comments": {
        // comments applicable to yaml files.
        "lineComment": "#",
        // comments applicable to tpl files.
        "blockComment": [ "{{- /*\n", "\n*/ -}}" ]
    },
    // symbols used as brackets
    "brackets": [
        ["{{", "}}"],
        ["{{-", "-}}"],
        ["{", "}"],
        ["[", "]"],
        ["(", ")"]
    ],
    // symbols that are auto closed when typing
    "autoClosingPairs": [
        ["{", "}"],
        ["{{-", "}}"],
        ["[", "]"],
        ["(", ")"],
        ["\"", "\""],
        ["'", "'"],
        ["`", "`"]
    ],
    // symbols that that can be used to surround a selection
    "surroundingPairs": [
        ["{{", "}}"],
        ["{{-", "}}"],
        ["{{-", "-}}"],
        ["{", "}"],
        ["[", "]"],
        ["(", ")"],
        ["\"", "\""],
        ["'", "'"],
        ["`", "`"]
    ]
}
```

My personal experience with Helm templating is that it's easier to manage if you do not default to chomping the trailing newline and only chomp leading whitespace ie `{{- blah }}` since the usual intention is to remove the template line if it ends up not being rendered (if, with, range etc).

**Pull request merged**